From ffc6f68fbda7190a8bbb02bfab110eb6f74d6e92 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Thu, 11 Jun 2020 10:31:08 -0400
Subject: gcc plugin ada use def

When GCC plugin is enabled "export-all-symbols" results in too many symbols.
That causes Ada build error. So, use gnat1.exe.def to reduce symbols.

---
 gcc/ada/gcc-interface/Make-lang.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/ada/gcc-interface/Make-lang.in b/gcc/ada/gcc-interface/Make-lang.in
index acbe2b877ca..ab0cbd6be07 100644
--- a/gcc/ada/gcc-interface/Make-lang.in
+++ b/gcc/ada/gcc-interface/Make-lang.in
@@ -44,6 +44,8 @@ RM = rm -f
 RMDIR = rm -rf
 
 
+COMMA := ,
+
 # Extra flags to pass to recursive makes.
 COMMON_ADAFLAGS= -gnatpg
 ifeq ($(TREECHECKING),)
diff --git a/gcc/ada/gcc-interface/Make-lang.in b/gcc/ada/gcc-interface/Make-lang.in
index acbe2b877ca..fd4956e4218 100644
--- a/gcc/ada/gcc-interface/Make-lang.in
+++ b/gcc/ada/gcc-interface/Make-lang.in
@@ -670,7 +670,7 @@ ada/libgnat/s-excmac.adb: $(srcdir)/ada/libgnat/s-excmac__$(EH_MECHANISM).adb
 #  stamp target in the parent directory whenever gnat1 is rebuilt
 gnat1$(exeext): $(TARGET_ADA_SRCS) $(GNAT1_OBJS) $(ADA_BACKEND) libcommon-target.a $(LIBDEPS)
 	+$(GCC_LLINK) -o $@ $(GNAT1_OBJS) $(ADA_BACKEND) \
-	  libcommon-target.a $(LIBS) $(SYSLIBS) $(BACKENDLIBS) $(CFLAGS)
+	  libcommon-target.a $(LIBS) $(SYSLIBS) $(subst -Wl$(COMMA)--export-all-symbols,../../gnat1$(exeext).def,$(BACKENDLIBS)) $(CFLAGS)
 	$(RM) stamp-gnatlib2-rts stamp-tools
 
 gnatbind$(exeext): ada/b_gnatb.o $(CONFIG_H) $(GNATBIND_OBJS) ggc-none.o libcommon-target.a $(LIBDEPS)
-- 
